samples/bluetooth/central_multilink/src/central_multilink.c

```diff
@@ -186,7 +187,7 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 
 	bt_conn_unref(conn);
 
-	if ((conn_count == 1U) && is_disconnecting) {
+	if ((conn_count == 1U) && (is_disconnecting || (reason == BT_HCI_ERR_CONN_FAIL_TO_ESTAB))) {
```

Comment:
Doesn't the `reason == BT_HCI_ERR_CONN_FAIL_TO_ESTAB` come in the `connected` callback, and not the disconnected callback?

Reply:
First connection_complete with reason success comes in `connected` callback, then 6 connection interval later `reason == BT_HCI_ERR_CONN_FAIL_TO_ESTAB` come in the `disconnected` callback.

I have added new two variants with 20 connections each.

In the many central devices tests, each central tries connecting at the same time to the single advertising peripheral. In bsim, one of 20 central manages to successful get its `CONNECT_IND` PDU to the peripheral, 19 others get "failed to establish" .... this goes on till all 20 centrals make 20 parallel connection to the single peripheral device.

The update in this line is to make the central device to re-scan and connect when it is the unlucky one amongst all centrals that sent the overlapping `CONNECT_IND`.